### PR TITLE
Fix inline script parsing issues

### LINKED ISSUE #572: Script inline constructor is flakey
Tried generating a script template with something like this:
```python
@script()
def echo(message: str) -> None:  # pragma: no cover
    """Prints out a message."""
    print(message)
```
and got something like this:
```yaml
source: "import os\nimport sys\nsys.path.append(os.getcwd())\nimport json\n\
        try: message = json.loads(r'''{{inputs.parameters.message}}''')\nexcept: message\
        \ = r'''{{inputs.parameters.message}}'''\n\ndef echo(message: str) -> None:\
        \  # pragma: no cover\n    \"\"\"Prints out a message.\"\"\"\n    # TODO Remove\
        \ function or replace with your logic\n    print(message)\n"
```

This seems to be because of the reliance on the function signature ending with a `:\n` string, from the code [here](https://github.com/argoproj-labs/hera/blob/main/src/hera/workflows/script.py#L334).

### PR DESCRIPTION
**Pull request by samj1912**
Fixes #572 

The fix makes the function parser a bit more robust by round-tripping through ast parse/unparse.

This ensures we get consistently formatted code each time and removes any extraneous comments which were previously causing parsing issues.

### CODE REVIEW

File: docs/examples/workflows/coinflip.md [hunk 1]
```diff
@@ -72,12 +72,9 @@
 
             import random
 
+            result = ''heads'' if random.randint(0, 1) == 0 else ''tails''
```
Comment (elliotgunton) [inline docs/examples/workflows/coinflip.md:75]: The double single quotes makes me nervous, does this run in python?

Comment (samj1912) [reply docs/examples/workflows/coinflip.md:75]: It's yaml first, if you convert it to json, the double single quotes go away.

**Review by elliotgunton — APPROVED** [review]
LGTM

**Review by flaviuvadan — APPROVED** [review]
🚀

**Pull request merged**